kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -454,6 +479,11 @@ private static StructField assignColumnIdAndPhysicalNameToField(
     if (!hasPhysicalName(field)) {
       // re-use old display names as physical names when a table is updated
       String physicalName = isNewTable ? "col-" + UUID.randomUUID() : field.getName();
+      if (useColumnIdForPhysicalName) {
+        long columnId = getColumnId(field);
+        physicalName = String.format("col-%s", columnId);
+      }
```

Comment:
Not sure if we have a style/convention with not having `null`, but it would be more efficient to do:
```suggestion
      String physicalName;
      if (useColumnIdForPhysicalName) {
        long columnId = getColumnId(field);
        physicalName = String.format("col-%s", columnId);
      } else {
        physicalName = isNewTable ? "col-" + UUID.randomUUID() : field.getName();
      }
```

Reply:
I think this is fine to do